src/plugins/unified_search/public/dataview_picker/data_view_picker.tsx

```diff
@@ -137,6 +139,7 @@ export const DataViewPicker = ({
       onSaveTextLanguageQuery={onSaveTextLanguageQuery}
       onTextLangQuerySubmit={onTextLangQuerySubmit}
       textBasedLanguage={textBasedLanguage}
+      shouldShowTextBasedLanguageTransitionModal={shouldShowTextBasedLanguageTransitionModal}
```

Comment:
Would be great to also hide the warning icon if the transition does not require a confirmation.

<img width="386" alt="Screenshot 2024-02-21 at 11 41 35" src="https://github.com/elastic/kibana/assets/1415710/40346f33-a75a-4923-ac99-815e008aa510">

Reply:
Yes this make sense! Added here: https://github.com/elastic/kibana/pull/177107/commits/def718936c39ef0bd13888fe05cadcc5f0947c62.